### DISCUSSION TITLE
Rewrites + Link doesn't redirect at client-side

### DISCUSSION THREAD

**Discussion by franklinjavier** [vercel/next.js#11687] [Help] [answered]
Our application renders a tracking page at two URLs
https://meurastreio.blzlocal.com.br/24101036
https://meurastreio.blzlocal.com.br/tracking/24101036

When I submit this form https://meurastreio.blzlocal.com.br, I'm trying to redirect (client side) but it doesn't work.

Rewrites config
```js
const nextConfig = {
  experimental: {
    async rewrites() {
      return [{ source: '/:id([0-9]+)', destination: '/tracking/:id' }]
    },
  }
}
```

```jsx
<form
  onSubmit={e => {
    e.preventDefault()
    setLoading(true)
    Router.push('/[id]', `/${orderNumber}`)
  }}
>
```
 
I tried ```Router.push('/tracking/[id]', `/tracking/${orderNumber}`)```, But it is preferable to use without the prefix.

Could anyone help?

**Comment by timneutkens** [comment, marked as answer]
This is expected as per the RFC https://github.com/zeit/next.js/issues/9081

**Reply by franklinjavier** [reply]
Yes, but It would be nice to push client-side route to /[id] instead /tracking/[id].

**Reply by Timer** [reply]
RFC to change this behavior is in #8207
